### PR TITLE
Button action for user prompt

### PR DESCRIPTION
**Pull request by FlaminWrap**
## Description
Added Prompt User action for buttons which acts as a dialog prompting the user to confirm or cancel before continuing.

Addresses: 
#7762

## App Export
[PromptUserApp-export-1668389698129.tar.gz](https://github.com/Budibase/budibase/files/9998824/PromptUserApp-export-1668389698129.tar.gz)

## Screenshots
Showing in the list of actions
image

After selecting prompt user
image

Custom title and message
image

Video
Link to video

### CODE REVIEW

**Review by melohagan — APPROVED** [review]
LGTM! Thanks for your contribution!

File: packages/client/src/utils/buttonActions.js [hunk 1]
```diff
@@ -429,12 +440,14 @@ export const enrichButtonActions = (actions, context) => {
...
                   const newContext = { ...context, actions: buttonContext }
 
-                  // Enrich and call the next button action
-                  const next = enrichButtonActions(
-                    actions.slice(i + 1),
-                    newContext
-                  )
-                  resolve(await next())
+                  // Enrich and call the next button action if there is more than one action remaining
+                  if (actions.length > 1) {
```
Comment (aptkingston) [inline packages/client/src/utils/buttonActions.js:444]: We don't actually need to add this check - `enrichButtonActions` won't do anyting when an empty array is passed in.

File: packages/client/src/utils/buttonActions.js [hunk 2]
```diff
@@ -327,6 +327,11 @@ const showNotificationHandler = action => {
   notificationStore.actions[type]?.(message, autoDismiss)
 }
 
+const promptUserHandler = action => {
```
Comment (aptkingston) [inline packages/client/src/utils/buttonActions.js:330]: We can just make this a completely empty function:
`const promptUserHandler = () => {}`

Doesn't affect functionality of course but may as well keep it as tidy as possible!

**Review by aptkingston — APPROVED** [review]
Thanks for the contribution @FlaminWrap! It's clever as well to just hook in to the existing confirmation prompt flow to handle this - keeps the change super tiny and safe.

Couple of suggestions just to remove redundant code, but functionality-wise it's perfect!

**Pull request merged**